sound/soc/sof/pm.c

```diff
@@ -433,12 +433,58 @@ EXPORT_SYMBOL(snd_sof_set_d0_substate);
 
 int snd_sof_resume(struct device *dev)
 {
+	struct snd_sof_dev *sdev = dev_get_drvdata(dev);
+	int ret;
+
+	if (sdev->s0_suspend && sof_ops(sdev)->set_power_state) {
```

Comment:
forgot to remove te check for set_power_state @keyonjie ?

Reply:
yep, thanks for spotting it, updating now.